### PR TITLE
Add support for Azure Functions

### PR DESCRIPTION
**Pull request by trstringer**
##### SUMMARY

These new modules (mutation and facts modules) add support for managing basic Azure Functions scenarios.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

- azure_rm_functionapp
- azure_rm_functionapp_facts

##### ANSIBLE VERSION

```
ansible 2.4.0 (azure-functions cd448ce758) last updated 2017/08/23 11:51:47 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/trstringer/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/trstringer/dev/ansible-dev/ansible/lib/ansible
  executable location = /home/trstringer/dev/ansible-dev/ansible/bin/ansible
  python version = 3.6.2 (default, Aug 11 2017, 11:59:59) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]
```

##### ADDITIONAL INFORMATION

Add support for Azure Functions.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -87,3 +87,4 @@ htmlcov/
 .coverage
 # ansible-test coverage results
 test/units/.coverage.*
+/test/integration/cloud-config-azure.yml
```
Comment (trstringer) [inline .gitignore:93]: Curious what the Ansible team thinks of adding this to gitignore. Obviously this is something we don't want to commit. I'm curious what the git handling for `cloud-config-aws.yml` is. If this is not desired, let me know and I can remove this change from the PR. I guess the alternative is adding this global gitignore.

**Comment by trstringer** [comment]
What do I need to do to exclude python2.6 from being tested? From my understanding, we didn't need to support python2.6 in the Azure modules. And I don't believe we've run into this issue with prior Azure module PRs. Am I missing something in my PR that would exclude python2.6?

**Comment by mattclay** [comment]
@tstringer All modules must have valid syntax under Python 2.6, 2.7, 3.5 and 3.6, even if they do not support running on all those versions (due to dependencies not supporting those versions). This is necessary for things like ansible-doc to work correctly.

The integration tests themselves will not be run on Python 2.6.

**Comment by mattclay** [comment]
CI failure in integration tests:

```
{
"changed": false,
"failed": true,
"msg": "The storage account named azfunccistor4 is already taken. - Reason.already_exists"
}
```

Make sure to use dynamically created names for resources which must be unique. Otherwise parallel runs of tests (such as above) will fail.

**Comment by nitzmahone** [comment]
rebuild_merge

**Comment by nitzmahone** [comment]
rebuild_merge

**Comment by wilsonge** [comment]
@tstringer I've just been reviewing this for a project, and looks like it's missing the ability to select a plan (either an App Service plan or a consumption plan) as part of the init.

Equivalent CLI commands would be:
`az functionapp create --name {{ app }} --storage-account {{ storage_account }} --resource-group {{ resource_group }} --plan {{ app_plan }}`

`az functionapp create -g {{ resource_group }} -n {{ app }} -s {{ storage_account }} -u {{ git_repo_uri }} --consumption-plan-location {{ location }} `

Is there a plan for this, would you like to to open a fresh issue in ansible or a mystery 3rd option :P

**Comment by trstringer** [comment]
@wilsonge yes that's correct, this module currently doesn't support app service plans. We're also working on adding the ability to create/configure app service plans through Ansible, and at that point I'll backport the functionality to specify an app service plan when you create a Function App.

Thanks! 👍 😄

**Pull request merged**